StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -369,7 +369,22 @@ extension PaymentSheet {
                     linkAccount.createPaymentDetails(with: paymentMethodParams) { result in
                         switch result {
                         case .success(let paymentDetails):
-                            confirmWithPaymentDetails(linkAccount, paymentDetails)
+                            if intent.linkPassthroughModeEnabled {
+                                // If passthrough mode, share payment details
+                                linkAccount.sharePaymentDetails(id: paymentDetails.stripeID) { result in
+                                    switch result {
+                                    case .success(let shareResponse):
+                                        confirmWithPaymentMethod(STPPaymentMethod(stripeId: shareResponse.paymentMethod))
+                                    case .failure(let error):
+                                        assertionFailure("Failed to share payment details \(error)")
+                                        // If this fails, confirm directly
+                                        confirmWithPaymentMethodParams(paymentMethodParams)
```

Comment:
It feels weird that we have an assertionFailure here and then attempt to confirmWithPaymentMethodParams... if this is really the intended behavior, then maybe we should add some metrics w/ an alarm?

Reply:
I kind of disagree with the assert here too but I'm following the existing code. Planned to clean it up and switch to alarms in MOBILESDK-1459